packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -625,6 +625,59 @@ where
         }
         .encode_to_vec())
     }
+
+    #[tracing::instrument(skip_all)]
+    async fn update_client(&self, dst_client_id: String) -> Result<Vec<u8>> {
+        let ethereum_client_state = self.ethereum_client_state(dst_client_id.clone()).await?;
+        let finality_update = self.beacon_api_client.finality_update().await?;
+        let latest_finalized_block_number =
+            finality_update.data.finalized_header.execution.block_number;
+
+        if latest_finalized_block_number <= ethereum_client_state.latest_execution_block_number {
+            tracing::warn!(
+                "No updates needed for client {}: latest finalized block number: {}, latest execution block number: {}",
+                dst_client_id,
+                latest_finalized_block_number,
+                ethereum_client_state.latest_execution_block_number
+            );
+
+            return Err(anyhow::anyhow!(
+                "No updates needed for client {}: latest finalized block number: {}, latest execution block number: {}",
+                dst_client_id,
+                latest_finalized_block_number,
+                ethereum_client_state.latest_execution_block_number
+            ));
+        }
+
+        tracing::info!(
+            "Generating tx to update client from block number: {} to block number: {}",
```

Comment:
For this, it has also been useful in the past to know how many updates and if the update is a period change or not (we log this in the relay_events call).

Reply:
addressed

Reply:
Great! Thanks 🙌